whylog/config/__init__.py

```diff
@@ -125,8 +126,19 @@ def _load_rules(self):
         ]
 
     def _load_log_types(self):
-        #TODO fix log types loading
-        pass
+        matchers = defaultdict(list)
+        matcher_definitions = self._load_file_with_config(self._log_type_path)
+        for definition in matcher_definitions:
+            if definition['matcher_class_name'] == 'RegexFilenameMatcher':
```

Comment:
Refactor this to route data to the right class using a dictionary <matcher_class_name, factory> instead of bunch of if statements in a for loop